src/main/java/io/fusionauth/samlv2/service/DefaultSAMLv2Service.java

```diff
@@ -701,7 +701,11 @@ <T> String buildPostRequest(JAXBElement<T> object, Class<T> type, boolean sign,
       Element toSign = document.getDocumentElement();
       String xml;
       if (sign) {
-        xml = signXML(privateKey, certificate, algorithm, xmlSignatureC14nMethod, document, toSign, null, includeKeyInfo);
+        // - The next sibling of the 'Issuer' may be null, this will cause the Signature to be inserted as the last element
+        //   of the assertion which is what we want.
+        Node issuer = toSign.getElementsByTagName("Issuer").item(0);
```

Comment:
This is currently used for `AuthenticationRequest`, `LogoutRequest`, and `LogoutResponse`.

Do we know if the issuer is required for all of these types? 

We have this same assumption in the `buildAuthnResponse` but there we document the element is required by spec. 

We should confirm, and if it is always required for these 3 use cases, we should document that here, and if not, we'll need to be defensive in case the element does not exist.

Reply:
It's weird that `LogoutResponse` uses the `buildPostRequest` method, but all of those types start with `Issuer` followed by `Signature` according to spec, and all of them will have the `Issuer` element based on the `toAuthnRequest()`, `toLogoutRequest()`, and `toLogoutResponse()` implementations.

Reply:
I see now. The spec does not require an `Issuer` for any of those even though we will always have it. I adjusted to use the same logic as signing the `ResponseType`, which does not require an `Issuer` and updated the comment to reflect the types that are signed here.